packages/react/src/FilteredActionList/FilteredActionListWithModernActionList.tsx

```diff
@@ -174,13 +198,15 @@ export function FilteredActionList({
                         {group.header?.title ? group.header.title : `Group ${group.groupId}`}
                       </ActionList.GroupHeading>
                       {getItemListForEachGroup(group.groupId).map((item, index) => {
-                        return <MappedActionListItem key={index} {...item} renderItem={listProps.renderItem} />
+                        const key = ('key' in item ? item.key : undefined) ?? item.id?.toString() ?? index.toString()
```

Comment:
```suggestion
                        const key = item.key ?? item.id?.toString() ?? index.toString()
```

I think these are equivalent